src/osd/OSD.cc

```diff
@@ -1451,11 +1449,11 @@ bool OSDService::prepare_to_stop()
 
 void OSDService::got_stop_ack()
 {
-  std::lock_guard l(is_stopping_lock);
+  std::scoped_lock l(is_stopping_lock);
```

Comment:
@tchaikov I took the opportunity to move to `scoped_lock` here, but I can revert if you think that's not necessary.